app/src/main/java/com/example/android/sunshine/app/DetailFragment.java

```diff
@@ -152,23 +157,22 @@ public void onActivityCreated(Bundle savedInstanceState) {
 
     void onLocationChanged(String newLocation) {
         // replace the uri, since the location has changed
-        Uri uri = mUri;
+        Uri uri = this.uri;
         if (null != uri) {
             long date = WeatherContract.WeatherEntry.getDateFromUri(uri);
-            Uri updatedUri = WeatherContract.WeatherEntry.buildWeatherLocationWithDate(newLocation, date);
-            mUri = updatedUri;
+            this.uri = WeatherEntry.buildWeatherLocationWithDate(newLocation, date);
             getLoaderManager().restartLoader(DETAIL_LOADER, null, this);
         }
     }
 
     @Override
     public Loader<Cursor> onCreateLoader(int id, Bundle args) {
```

Comment:
Do we have to use `Loader` and `Cursor` in this project? Not familiar with it so not sure if this is your choice or Udacity's choice. If we do need to use `Loader` and `Cursor`, could we move these methods onto a different class that the fragment could then instantiate? Right now there is a lot of logic in the fragment itself.